src/pytest_mypy/__init__.py

```diff
@@ -253,6 +267,14 @@ def runtest(self) -> None:
         """Raise a MypyError if mypy exited with a non-zero status."""
         results = MypyResults.from_session(self.session)
         if results.status:
+            if self.session.config.option.mypy_xfail:
+                self.add_marker(
+                    pytest.mark.xfail(
+                        raises=MypyError,
+                        reason=f"mypy's non-zero exit status ({results.status})"
+                        " was xfailed by --mypy-xfail.",
+                    )
+                )
```

Comment:
Nit: This code feels a little redundant with the earlier code. Maybe extract a method?

Reply:
Think I'm gonna punt on this for now... I've got a branch that explores using different errors for these checks, and that would change what gets passed to `raises` (which I think would make a method less useful).